### ISSUE TITLE
SWIM - An additional option to set a starter list of nodes to avoid multicast

### ISSUE THREAD

**Issue by zarusz** [zarusz/SlimCluster#7]
Currently, when a Node joins it multicast the join to let other members it had joined.
Other members then send a welcome message with the current list of members, so that the Node can catch up on the cluster membership fast.

However, we could also provide an option where the initial list of few members is provided upfront as a configuration (bootstrap node list). This could work in some scenarios and thus we could avoid the reliance on the multicast for the initial discovery.

**Comment by Ephron-WL** [comment]
Some virtual networks running in software defined networking may not support multicast.  So, eliminating multicast would be a good idea.

**Comment by zarusz** [comment]
Yes, indeed. Thanks for the feedback!

**Comment by Ephron-WL** [comment]
Do you think maybe a DNS lookup could be used to return a list of member addresses for unicast notification?

**Comment by zarusz** [comment]
Yes, that strategy could be added to the library too.  Ideally if we have few strategies to select from for users as each will have a different setup and use case.

Is this something you would need right now? I'm curious what setup/environment you are targeting.

Also, happy to accept a contribution.

**Comment by Ephron-WL** [comment]
I don’t need it right now.  I’d be willing to contribute the feature.
